appsec/src/helper/subscriber/waf.cpp

```diff
@@ -417,11 +417,10 @@ void instance::listener::call(
     dds::parameter_view &data, event &event, const std::string &rasp_rule)
 {
     ddwaf_result res;
-    DDWAF_RET_CODE code;
     auto run_waf = [&]() {
         dds::parameter_view *persistent = rasp_rule.empty() ? &data : nullptr;
         dds::parameter_view *ephemeral = rasp_rule.empty() ? nullptr : &data;
-        code = ddwaf_run(
+        code_ = ddwaf_run(
```

Comment:
This code will be replaced on every run, if one has an error (for some reason), and the next one succeeds, you'll send the success code instead of the error.

Reply:
You need to conceptually store a map from error code to a count instead. You can ignore DDWAF_OK and DDWAF_MATCH.

If there is more than one error code, you'll submit multiple metric counts.

Reply:
Also this should apply only to rasp runs

Reply:
Sorted, I got confused with the part of the document talking about the span tag instead

Reply:
The error code still has to be reported as a tag in non-RASP calls using `appsec.waf.error`.

Reply:
> The error code still has to be reported as a tag in non-RASP calls using `appsec.waf.error`.

That one will be done on the non SLO ones